contracts/cw721-metadata-onchain/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+/// There are 2 possibiities for using metadata on chain:
+///
+/// cw721-metadata-onchain:
+///
+/// ```rust
+/// // instantiate:
+/// let contract = Cw721MetadataContract::default();
+/// let info = mock_info(CREATOR, &[]);
+/// let init_msg = Cw721InstantiateMsg {
+///     name: "SpaceShips".to_string(),
+///     symbol: "SPACE".to_string(),
+///     collection_info_extension: None,
+///     minter: None,
+///     creator: None,
+///     withdraw_address: None,
+/// };
+/// // ...
+/// // mint:
+/// let token_id = "Enterprise";
+/// let token_uri = Some("https://starships.example.com/Starship/Enterprise.json".into());
+/// let extension = Some(NftExtensionMsg {
+///     description: Some("description1".into()),
+///     name: Some("name1".to_string()),
+///     attributes: Some(vec![Trait {
+///         display_type: None,
+///         trait_type: "type1".to_string(),
+///         value: "value1".to_string(),
+///     }]),
+///     ..NftExtensionMsg::default()
+/// });
+/// let exec_msg = ExecuteMsg::Mint {
+///     token_id: token_id.to_string(),
+///     owner: "john".to_string(),
+///     token_uri: token_uri.clone(),
+///     extension: extension.clone(),
+/// };
+/// // ...
+/// ```
+///
+/// cw721-base with metadata onchain:
+/// ```rust
+/// // instantiate:
+/// let contract = Cw721Contract::<
+///     DefaultOptionalNftExtension, // use `Option<Empty>` for no nft metadata
+///     DefaultOptionalNftExtensionMsg, // use `Option<Empty>` for no nft metadata
+///     DefaultOptionalCollectionExtension, // use `Option<Empty>` for no collection metadata
+///     DefaultOptionalCollectionExtensionMsg, // use `Option<Empty>` for no collection metadata
+///     Empty,
+///     Empty,
+///     Empty,
+/// >::default();
+/// let info = mock_info(CREATOR, &[]);
+/// let init_msg = Cw721InstantiateMsg {
+///     name: "SpaceShips".to_string(),
+///     symbol: "SPACE".to_string(),
+///     collection_info_extension: None,
+///     minter: None,
+///     creator: None,
+///     withdraw_address: None,
+/// };
+/// //...
+/// // mint:
+/// let token_id = "Enterprise";
+/// let token_uri = Some("https://starships.example.com/Starship/Enterprise.json".into());
+/// let extension = Some(NftExtensionMsg {
+///     description: Some("description1".into()),
+///     name: Some("name1".to_string()),
+///     attributes: Some(vec![Trait {
+///         display_type: None,
+///         trait_type: "type1".to_string(),
+///         value: "value1".to_string(),
+///     }]),
+///     ..NftExtensionMsg::default()
+/// });
+/// let exec_msg = Cw721ExecuteMsg::<
+///     DefaultOptionalNftExtensionMsg,
+///     DefaultOptionalCollectionExtensionMsg,
+///     Empty,
+/// >::Mint {
+///     token_id: token_id.to_string(),
+///     owner: "john".to_string(),
+///     token_uri: token_uri.clone(),
+///     extension: extension.clone(), // use `extension: None` for no metadata
+/// };
+/// //...
+/// ```
+pub type Cw721MetadataContract<'a> = cw721_base::Cw721Contract<
```

Comment:
`cw721-metadata-onchain` is back again. Checkout above docs which shows the differene and how to use `cw721-metadata-onchain` and `cw721-base` with onchain metadata (aka `NftExtension`).

@shanev @humanalgorithm